src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/BrowserWebSocket.cs

```diff
@@ -33,122 +34,126 @@ public override WebSocketState State
         {
             get
             {
-#if FEATURE_WASM_THREADS
-                lock (_thisLock)
+                lock (_lockObject)
                 {
-#endif
-                    if (_innerWebSocket == null || _disposed || (_state != WebSocketState.Connecting && _state != WebSocketState.Open && _state != WebSocketState.CloseSent))
+                    if (_innerWebSocket == null || _disposed || _state == WebSocketState.Aborted || _state == WebSocketState.Closed)
                     {
                         return _state;
                     }
-#if FEATURE_WASM_THREADS
-                } //lock
-#endif
-
-#if FEATURE_WASM_THREADS
-                return FastState = _innerWebSocket!.SynchronizationContext.Send(static (BrowserWebSocket self) =>
-                {
-                    lock (self._thisLock)
+                    var st = GetReadyState(_innerWebSocket!);
```

Comment:
this call is synchronously remoted to the thread that owns the socket, right?

Reply:
right, synchronously in sense, that the sender waits for the WS thread to read and respond. There could be other un-ordered actions from other threads, sending into same WS thread.

I think that I could afford to call it from locked section, because it doesn't make call any other action inside, which could lead to deadlock.